scrapy_poet/downloadermiddlewares.py

```diff
@@ -107,6 +108,19 @@ def process_response(
         and an injectable attribute,
         the user-defined ``cb_kwargs`` takes precedence.
         """
+        if request.callback is None:
+            callback = get_callback(request, spider)  # should return parse()
+            response_annotation = get_response_annotation(callback)
+
+            if issubclass(response_annotation.annotation, DummyResponse):
+                warnings.warn(
+                    "A request has been encountered with callback=None which "
+                    "defaults to the parse() method. On such cases, when the "
+                    "parse() method is annotated with DummyResponse, "
```

Comment:
It's not about DummyResponse only, right?

Reply:
Do you on its subclasses as well? Good catch! Updated in https://github.com/scrapinghub/scrapy-poet/pull/106/commits/59076de43c8d71e58dcedeb58a07bb7dc3c0a18a

Reply:
What we discussed off GitHub was to discourage the use of `parse` as a callback that gets injected stuff, and to warn if `parse` has _anything_ that would require injection, be it DummyResponse, HttpResponse, or anything else.

By the way, to do it properly, I am thinking that we need to ensure we do it not only for callback=None, but also for callback='parse' and callback=spider.parse.

Reply:
As @Gallaecio said, it's about anything we cn provide, not just DummyResponse.

> By the way, to do it properly, I am thinking that we need to ensure we do it not only for callback=None, but also for callback='parse' and callback=spider.parse.

Why would callback=spider.parse or "parse" be an issue? If user wants to use this callback explicitly, would it be possible to provide the dependencies, without any warnings?

Reply:
OK, so we would not be warning against the use of resolvable parameters in `parse` _in general_, but only when there is at least one detected use of callback=None. Works for me.

Reply:
> OK, so we would not be warning against the use of resolvable parameters in parse in general, but only when there is at least one detected use of callback=None. Works for me.

Yes + only if there are any annotated parameters in the `parse()` method.

Thanks for clarifying this gents!